tool/generator/lib/src/projection/utils.dart

```diff
@@ -341,6 +345,217 @@ String parseGenericTypeIdentifierName(TypeIdentifier typeIdentifier) {
+        parseTypeIdentifierSignature(typeIdentifier.typeArg!);
+    final secondArgSignature =
+        parseTypeIdentifierSignature(typeIdentifier.typeArg!.typeArg!);
+    return 'pinterface($parentTypeGuid;$firstArgSignature;$secondArgSignature)';
+  }
+
+  return 'pinterface($parentTypeGuid;${parseTypeIdentifierSignature(typeIdentifier.typeArg!)})';
+}
+
+const wrtPinterfaceNamespace = [
+  0x11,
+  0xf4,
+  0x7a,
+  0xd5,
+  0x7b,
+  0x73,
+  0x42,
+  0xc0,
+  0xab,
+  0xae,
+  0x87,
+  0x8b,
+  0x1e,
+  0x16,
+  0xad,
+  0xee,
+];
+
+/// Returns the IID for the given [signature].
+///
+/// To learn more about this, please see https://learn.microsoft.com/en-us/uwp/winrt-cref/winrt-type-system#guid-generation-for-parameterized-types
+String iidFromSignature(String signature) {
+  final signatureInBytes = const Utf8Encoder().convert(signature);
+  final data = [...wrtPinterfaceNamespace, ...signatureInBytes];
+  final sha1Bytes = sha1.convert(data).bytes;
+
+  final first = Int8List(4)
+    ..[0] = sha1Bytes[0]
+    ..[1] = sha1Bytes[1]
+    ..[2] = sha1Bytes[2]
+    ..[3] = sha1Bytes[3];
+  final firstNumber = first.buffer.asByteData().getUint32(0);
+
+  final second = Int8List(2)
+    ..[0] = sha1Bytes[4]
+    ..[1] = sha1Bytes[5];
+  final secondNumber = second.buffer.asByteData().getUint16(0);
+
+  final third = Int8List(2)
+    ..[0] = sha1Bytes[6]
+    ..[1] = sha1Bytes[7];
+  var thirdNumber = third.buffer.asByteData().getUint16(0);
+  thirdNumber = (thirdNumber & 0x0fff) | (5 << 12);
+
+  final fourthNumber = (sha1Bytes[8] & 0x3f) | 0x80;
+
+  final guidChars = List<int>.filled(38, 0);
+  var offset = 0;
+  // {dddddddd-dddd-dddd-dddd-dddddddddddd}
+  guidChars[offset++] = '{'.codeUnitAt(0);
+  offset =
+      _hexsToChars(guidChars, offset, firstNumber >> 24, firstNumber >> 16);
+  offset = _hexsToChars(guidChars, offset, firstNumber >> 8, firstNumber);
+  guidChars[offset++] = '-'.codeUnitAt(0);
+  offset = _hexsToChars(guidChars, offset, secondNumber >> 8, secondNumber);
+  guidChars[offset++] = '-'.codeUnitAt(0);
+  offset = _hexsToChars(guidChars, offset, thirdNumber >> 8, thirdNumber);
+  guidChars[offset++] = '-'.codeUnitAt(0);
+  offset = _hexsToChars(guidChars, offset, fourthNumber, sha1Bytes[9]);
+  guidChars[offset++] = '-'.codeUnitAt(0);
+  offset = _hexsToChars(guidChars, offset, sha1Bytes[10], sha1Bytes[11]);
+  offset = _hexsToChars(guidChars, offset, sha1Bytes[12], sha1Bytes[13]);
+  offset = _hexsToChars(guidChars, offset, sha1Bytes[14], sha1Bytes[15]);
+  guidChars[offset++] = '}'.codeUnitAt(0);
+
+  return String.fromCharCodes(guidChars).toUpperCase();
+}
+
+int _hexToChar(int a) {
+  a = a & 0xf;
+  return (a > 9) ? a - 10 + 0x61 : a + 0x30;
+}
+
+int _hexsToChars(List<int> guidChars, int offset, int a, int b) {
+  guidChars[offset++] = _hexToChar(a >> 4);
+  guidChars[offset++] = _hexToChar(a);
+  guidChars[offset++] = _hexToChar(b >> 4);
+  guidChars[offset++] = _hexToChar(b);
+  return offset;
+}
```

Comment:
These are adapted to Dart from the [System.Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=dotnet-uwp-10.0)'s toString() implementation https://referencesource.microsoft.com/#mscorlib/system/guid.cs,1278